src/server/api/routes/users/v2/nfts/featured.js

```diff
@@ -44,27 +63,67 @@ router.post(
       await walletUserCollection.doc(user).update({
         featuredNFTClassIds: FieldValue.arrayUnion(classId),
       });
+      publisher.publish(PUBSUB_TOPIC_MISC, req, {
+        logType: 'UserNFTClassFeatured',
+        user,
+        classId,
+      });
       res.sendStatus(200);
     } catch (err) {
       handleRestfulError(req, res, next, err);
     }
   }
 );
 
-router.delete(
-  '/:wallet/nfts/featured/:classId',
+router.post(
+  '/:wallet/nfts/hidden',
   authenticateV2Login,
   checkParamWalletMatch,
   async (req, res, next) => {
     try {
       setPrivateCacheHeader(res);
-      const { wallet: user, classId } = req.params;
+      const { wallet: user } = req.params;
+      const { classId } = req.body;
       if (!classId) {
         res.status(400).send('CLASS_ID_MISSING');
         return;
       }
       await walletUserCollection.doc(user).update({
         featuredNFTClassIds: FieldValue.arrayRemove(classId),
+        hiddenNFTClassIds: FieldValue.arrayUnion(classId),
+      });
+      publisher.publish(PUBSUB_TOPIC_MISC, req, {
+        logType: 'UserNFTClassHidden',
+        user,
+        classId,
+      });
+      res.sendStatus(200);
+    } catch (err) {
+      handleRestfulError(req, res, next, err);
+    }
+  }
+);
+
+router.post(
+  '/:wallet/nfts/unhidden',
```

Comment:
I think the above 3 API can be handled with a single path `'/:wallet/nfts/:state'`

There are 3 states, `featured`/`hidden`/`normal`